---
title: QC Formats
description: Practical Activity
slug: ale_tarea_01
date: 2025-02-09 
image: cover.jpg
categories:
    - Activity
    - Bioinformatics
    - Basic
tags:
    - FASTQC
    - File Managing
    - BCF
    - BAM
    - SAMTOOLS

---

## Practical BAM file

•  Using your VPN go to the cluster

• Using the file NA20538.bam located at /mnt/atgc-d1/bioinfoII/data/format_qc

• Answer the following questions

### What does RG stand for?

**Sgam**: It means read group, wich in turn indicates the number of reads for a specific sample. Its important to mention that when multiplexing in a single lane, each library or sample has its own read group. The relevance of having read groups from the same sample in different lanes and flow cells is to be able to discern which part of the error is due to the sequenciator or other circumnstances (batch effect).
Ref: <https://support.illumina.com/help/BS_App_MDProcessor_Online_1000000007932/Content/Source/Informatics/BAM-Format.htm>, <https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups>

**Ferro:** 
The @RG stand has this information: 

ID: Is a read group identifier. Each @RG line must have a unique ID.
The value of this ID is used in the RG tags of alignment record labels.
When merging the SAM files for collision control you can modify the IDs.
Ref: <https://support.illumina.com/help/BS_App_MDProcessor_Online_1000000007932/Content/Source/Informatics/BAM-Format.htm>, <https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups>
<https://samtools.github.io/hts-specs/SAMv1.pdf>

**Suazo:** 

```{bash, eval = FALSE}
samtools view -H NA20538.bam | grep '^@RG'
@RG	ID:ERR003612	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003761	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003762	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003763	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003764	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003765	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003766	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003767	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003811	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003812	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003813	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003814	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003815	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR003816	PL:ILLUMINA	LB:g1k-sc-NA20538-TOS-1	PI:2000	DS:SRP000540	SM:NA20538	CN:SC
@RG	ID:ERR015472	PL:ILLUMINA	LB:g1k-sc-NA20538-A	PI:450	DS:SRP000540	SM:NA20538	CN:SC

```

### What is the lane ID? ("Lane” is the basic independent run of a high- throughput sequencing machine. For Illumina machines, this is the physical sequencing lane. Reads from one lane are identified by the same read group ID and the information about lanes can be found in the header in lines starting with "@RG".)

We suppose that the Lane ID in this case is the tag ID, which does not repeat.  According to <https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups>, there could be multiplexing in each lane, and for that reason i'm confused with the definition that all samples of a lane are represented by the same RG.

### What is the sequencing platform?

Illumina, represented by the tag PL

> Hint: you want to use samtools. Files are huge so remember to pipe.
samtools view -H NA20538.bam | less -S

### What version of the human assembly was used to perform the alignments?

**Ferro:** 

### What programs were used to create this BAM file?

**Suazo:** 

```{bash, eval = F}
@PG     ID:GATK IndelRealigner  VN:1.0.4487     CL:SNPsFileForDebugging=null targetIntervals=/lustre/scratch105/projects/g1k/ref/broad_rec...
@PG     ID:GATK TableRecalibration      VN:1.0.4487     CL:output_bam=null window_size_nqs=5 force_read_group=null smoothing=1 default_pla...

```

The Programs used were GATK IndelRealigner and GATK TableRecalibration

### What version of bwa was used to align the reads?

**Suazo:**

```{bash, eval = F}
@PG     ID:bwa  VN:0.5.5
```

The version of bwa used was 0.5.5

### What is the name of the first read?

**Suazo:**

```{bash, eval = F}
 samtools view NA20538.bam | grep -m 1 "^ERR" | awk '{print $1}'
ERR003814.1408899
```

The name of the first read is ERR003814.1408899 

### What position does the alignment of the read start at?

**Suazo:**

```{bash, eval = F}
ERR003814.1408899       163     1       19999970        23      37M     =       20000147        213     GGTGGGTGGATCACCTGAGATCGGGAGTTTGAGA
```

The position of the alignment of the read start at 19999970


### What is the mapping quality of the first read?

**Suazo:**
```{bash, eval = F}
ERR003814.1408899       163     1       19999970        23      37M     =       20000147        213     GGTGGGTGGATCACCTGAGATCGGGAGTTTGAGA
```

The mapping quality of the first read is 23

## Practical bcf
> Now use file 1kg.bcf

### What is a bcf?
A bcf is like a vcf, but in binary, in which the fields are arranged for fast access. They store information about variants in read alignments.

### Can you convert bcf to vcf using bcftools? How?

### How many samples are in the BCF? Hint: use the -l option.

### What is the genotype of the sample HG00107 at the position 20:24019472? 
>Hint: use the combination of -r, -s, and -f '[ %TGT]\n' options.

### How many positions there are with more than 10 alternate alleles? (See the INFO/AC tag.) Hint: use the -i filtering option.

### List all positions where HG00107 has a non-reference genotype and the read depth is bigger than 10.

## Practical stats

> Using samtools stats answer

`samtools stats NA20538.bam | grep ^SN`


### What is the total number of reads?

**Suazo:**

```{bash, eval = F}
SN	raw total sequences:	347367	# excluding supplementary and secondary reads
```

### What proportion of the reads were mapped?

**Suazo:**

```{bash, eval = F}
SN	reads mapped:	323966
```

The proportion of the reads that were mapped is 323966/347367 = 0.932
### How many reads were mapped to a different chromosome?

```{bash, eval = F}
SN	pairs on different chromosomes:	4055
```

There were 4055 reads mapped to a different chromosome

### What is the insert size mean and standard deviation?

```{bash, eval = F}
SN	insert size average:	190.3
SN	insert size standard deviation:	136.4
```

The insert size mean is 190.3 and the standard deviation is 136.4

### Hint: options -F SECONDARY

> Get used to combine these commands with basic linux commands (|, grep, less, head, etc.)
