---
title: QC Formats
description: Practical Activity
slug: ale_tarea_01
date: 2025-02-09 
image: cover.jpg
categories:
    - Activity
    - Bioinformatics
    - Basic
tags:
    - FASTQC
    - File Managing
    - BCF
    - BAM
    - SAMTOOLS

---



<div id="practical-bam-file" class="section level2">
<h2>Practical BAM file</h2>
<p>• Using your VPN go to the cluster</p>
<p>• Using the file NA20538.bam located at /mnt/atgc-d1/bioinfoII/data/format_qc</p>
<p>• Answer the following questions</p>
<div id="what-does-rg-stand-for" class="section level3">
<h3>What does RG stand for?</h3>
<p><strong>Sgam</strong>: It means read group, wich in turn indicates the number of reads for a specific sample. Its important to mention that when multiplexing in a single lane, each library or sample has its own read group. The relevance of having read groups from the same sample in different lanes and flow cells is to be able to discern which part of the error is due to the sequenciator or other circumnstances (batch effect).
Ref: <a href="https://support.illumina.com/help/BS_App_MDProcessor_Online_1000000007932/Content/Source/Informatics/BAM-Format.htm" class="uri">https://support.illumina.com/help/BS_App_MDProcessor_Online_1000000007932/Content/Source/Informatics/BAM-Format.htm</a>, <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups" class="uri">https://gatk.broadinstitute.org/hc/en-us/articles/360035890671-Read-groups</a></p>
<p>Suazo:</p>
</div>
<div id="what-is-the-lane-id-lane-is-the-basic-independent-run-of-a-high--throughput-sequencing-machine.-for-illumina-machines-this-is-the-physical-sequencing-lane.-reads-from-one-lane-are-identified-by-the-same-read-group-id-and-the-information-about-lanes-can-be-found-in-the-header-in-lines-starting-with-rg." class="section level3">
<h3>What is the lane ID? (“Lane” is the basic independent run of a high- throughput sequencing machine. For Illumina machines, this is the physical sequencing lane. Reads from one lane are identified by the same read group ID and the information about lanes can be found in the header in lines starting with “<span class="citation">@RG</span>”.)</h3>
</div>
<div id="what-is-the-sequencing-platform" class="section level3">
<h3>What is the sequencing platform?</h3>
<blockquote>
<p>Hint: you want to use samtools. Files are huge so remember to pipe.
samtools view -H NA20538.bam | less -S</p>
</blockquote>
</div>
<div id="what-version-of-the-human-assembly-was-used-to-perform-the-alignments" class="section level3">
<h3>What version of the human assembly was used to perform the alignments?</h3>
</div>
<div id="what-programs-were-used-to-create-this-bam-file" class="section level3">
<h3>What programs were used to create this BAM file?</h3>
</div>
<div id="what-version-of-bwa-was-used-to-align-the-reads" class="section level3">
<h3>What version of bwa was used to align the reads?</h3>
</div>
<div id="what-is-the-name-of-the-first-read" class="section level3">
<h3>What is the name of the first read?</h3>
</div>
<div id="what-position-does-the-alignment-of-the-read-start-at" class="section level3">
<h3>What position does the alignment of the read start at?</h3>
</div>
<div id="what-is-the-mapping-quality-of-the-first-read" class="section level3">
<h3>What is the mapping quality of the first read?</h3>
</div>
</div>
<div id="practical-bcf" class="section level2">
<h2>Practical bcf</h2>
<blockquote>
<p>Now use file 1kg.bcf</p>
</blockquote>
<div id="what-is-a-bcf" class="section level3">
<h3>What is a bcf?</h3>
</div>
<div id="can-you-convert-bcf-to-vcf-using-bcftools-how" class="section level3">
<h3>Can you convert bcf to vcf using bcftools? How?</h3>
</div>
<div id="how-many-samples-are-in-the-bcf-hint-use-the--l-option." class="section level3">
<h3>How many samples are in the BCF? Hint: use the -l option.</h3>
</div>
<div id="what-is-the-genotype-of-the-sample-hg00107-at-the-position-2024019472" class="section level3">
<h3>What is the genotype of the sample HG00107 at the position 20:24019472?</h3>
<blockquote>
<p>Hint: use the combination of -r, -s, and -f ‘[ %TGT]’ options.</p>
</blockquote>
</div>
<div id="how-many-positions-there-are-with-more-than-10-alternate-alleles-see-the-infoac-tag.-hint-use-the--i-filtering-option." class="section level3">
<h3>How many positions there are with more than 10 alternate alleles? (See the INFO/AC tag.) Hint: use the -i filtering option.</h3>
</div>
<div id="list-all-positions-where-hg00107-has-a-non-reference-genotype-and-the-read-depth-is-bigger-than-10." class="section level3">
<h3>List all positions where HG00107 has a non-reference genotype and the read depth is bigger than 10.</h3>
</div>
</div>
<div id="practical-stats" class="section level2">
<h2>Practical stats</h2>
<div id="using-samtools-stats-answer" class="section level3">
<h3>Using samtools stats answer</h3>
</div>
<div id="what-is-the-total-number-of-reads" class="section level3">
<h3>What is the total number of reads?</h3>
</div>
<div id="what-proportion-of-the-reads-were-mapped" class="section level3">
<h3>What proportion of the reads were mapped?</h3>
</div>
<div id="how-many-reads-were-mapped-to-a-different-chromosome" class="section level3">
<h3>How many reads were mapped to a different chromosome?</h3>
</div>
<div id="what-is-the-insert-size-mean-and-standard-deviation" class="section level3">
<h3>What is the insert size mean and standard deviation?</h3>
</div>
<div id="hint-options--f-secondary" class="section level3">
<h3>Hint: options -F SECONDARY</h3>
<blockquote>
<p>Get used to combine these commands with basic linux commands (|, grep, less, head, etc.)</p>
</blockquote>
</div>
</div>
